core/source/Url.mint

```diff
@@ -66,6 +66,16 @@ module Url {
     `
   }
 
+  /*
+  Converts the given `Url` to a `String`.
+
+    (Url.parse("https://www.example.com/path/?search=foo#hash")
+    |> Url.toString()) == "https://www.example.com/path/?search=foo#hash"
+  */
+  fun toString (url : Url) : String {
+    `#{url}.origin + #{url}.path + #{url}.search + #{url}.hash`
```

Comment:
Why not use built in `toString()`?

```suggestion
    `#{url}.toString()`
```

Reply:
`Url` [is an object internally](https://github.com/mint-lang/mint/blob/26c370f786ea56e66d12d23df804304e506efa01/core/source/Url.mint#L55-L64), so calling `.toString()` on it would return `"[object Object]"`.

Reply:
I would guess, to return `"[object Object]"` would be the task of an `.repr` function, comparable to `repr` in Python. But even then, I wonder if `"[object Object]"` is a useful representation. Javascript has done this not so userfriendly in my opinion. Something like `JSON.stringify(url)` would have fitted more.

Reply:
Following code prints the url back 🤷‍♂️ 

```js
const url = new URL("https://github.com/mint-lang/mint/pull/603#discussion_r1204579853")
console.log(url.toString()) // => https://github.com/mint-lang/mint/pull/603#discussion_r1204579853
```

Reply:
Oh, right. The mint's `Url` is not backed by the JS' `URL`, my bad.

Reply:
> @gdotdesign Was the decision to decouple the `Url` from the JS counterpart was done on purpose?

Yes, `Url` in Mint is a record. Using the JS `URL` would mean it would be an opaque type and helper methods would have been needed for getting it's parts.